### DISCUSSION TITLE
Question on random crop "End-to-end optimized image compressio"

### DISCUSSION THREAD

**Discussion by Mareeta26** [tensorflow/compression#87] [Q&A] [answered]
Hi,
I'm relatively new to the data augmentation methods.I would like to know more the cropping done on the paper "End-to-end optimized image compression". 
What I understood is the images are randomly cropped to 256x256 patches and then fed to the training model. I tried to remove the cropping in training function, but couldn't train since the image was no longer 256x256.
How does it work for testing? I found that testing images are not cropped or resized. How does the trained model recognize the new sizes in the testing set?
Please advise

Thanks,
Mareeta

**Comment by lingyu98** [comment, marked as answer]
Hi Mareeta,

  In my understanding, the analysis transform is implemented by a _convolutional neural network_. This architecture utilizes a number of learnable filters to scan the input image and computes the output feature maps by convolution. The output size of each convolutional layer(including the size of the bottleneck layer) is determined by the size of the input layer and the striding length. In other words, the CNN architecture is adaptable to any (large enough) sized input. 

  During the training period, the input image sizes are required to be the same because you are training in batches, every image has to have the same height and widths so that it forms a 4D matrix before being fed into the network. However, images from most training sets come in all sizes. That's why cropping is necessary. You can change the patch size from 256x256 to any size you like, as long as you crop the images to the same size.

  In testing, we want to test the full image of each image from the test set so that results are comparable to certain standards. So there shouldn't be random cropping in the testing, since that would essentially change your test set, giving unreliable results. As mentioned, the CNN adapts to different sizes of inputs so there are no restraints on the size of test images. (Unless you want to test them in patches as well, which, empirically is faster)

Lingyu

**Reply by Mareeta26** [reply]
@Mikeyboiii Thank you very much for the reply.

**Reply by jonaballe** [reply]
Thanks for the accurate explanation!
